### PR TITLE
Support json & array types

### PR DESCRIPTION
**Pull request by jiaoew1991**
See also #23389 
/kind feature

### CODE REVIEW

**Comment by yah01** [comment]
Working on it

**Comment by yah01** [comment]
related #23389

**Comment by yah01** [comment]
rerun ut

File: internal/core/src/common/Column.h [hunk 1]
```diff
@@ -26,9 +27,13 @@
 #include "common/Utils.h"
 #include "exceptions/EasyAssert.h"
 #include "fmt/core.h"
+#include "log/Log.h"
+#include "nlohmann/json.hpp"
 
 namespace milvus::segcore {
 
+#define FIELD_DATA(info, field) (info->scalars().field##_data().data())
```
Comment (congqixia) [inline internal/core/src/common/Column.h:35]: Remove unused macro

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**